.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,51 @@
+name: Run benchmark for parser
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  # deployments permission to deploy GitHub pages website
+  deployments: write
```

Comment:
Just a thought: maybe writing to GitHub pages should only be enabled when running on the main branch? If I understand the benchmark action, then the benchmarks should only be updated when running on main, but I wonder if an additional guard would help.

Reply:
But still, it would be possible to modify the PR workflow and push results from PR (it just will be more obvious because the permissions will be modified)

Reply:
There is already an issue for that: https://github.com/Kotlin/kotlinx-benchmark/issues/185
`Re-declaration of dependencies.`